### ISSUE TITLE
[ui5-table-row] Table rows are invisible when added after initialization

### ISSUE THREAD

**Issue by brodo** [SAP/ui5-webcomponents#839]
**Describe the bug**
When `ui5-table-row`s are added to a `ui5-table` using Javascript, they are invisible.

**To reproduce**
1. Add a `ui5-table` to your HTML
2. Add some `ui5-table-row`s to the table using Javascript

**Isolated example**
Since Code Sandbox is not working with rc.3, I created a repo with contains a minimal example:
https://github.com/brodo/ui5-web-components-table-row-bug

**Expected behavior**
The rows should be visible

**Screenshots**
<img width="1337" alt="ui5-table-row-bug" src="https://user-images.githubusercontent.com/377911/66568840-82800700-eb6a-11e9-88eb-c3d081ad786a.png">

As you can see in the screenshot, the rows are there, they are just invisible. If I add divs instead of ui5-table-rows they are visible.

**Context**
- UI5 Web Components version: 1.0.0-rc.3
- OS/Platform: Mac
- Browser *(if relevant)*: Tested Firefox and Brave
- Other information: this might have been introduced when fixing #764 

**Affected components**
ui5-table-row

By the way, I use [Svelte](https://svelte.dev/) for my app and I encountered this bug when I updated to rc.3 and all my tables where gone...

**Comment by ilhan007** [comment]
Hello @brodo

you have 3 columns, but you are building rows with two cells, which causes a JS error.

If you remove the 3rd column or provide a 3rd cell for the row, it will work fine.

Still, we have to provide better solution for this case.

**Comment by brodo** [comment]
Thanks for looking into this @ilhan007 ! Should I keep this open for tracking the need for better error handling when the number of cells is incorrect? 

PS: I'm still investigating the problem in my app which this was related to. I have the right amount of cells there...

**Comment by brodo** [comment]
Hi! In order to debug why tables in my app stopped working after I updated to 1.0.0-rc.3 I've built another minimal example here: https://github.com/brodo/svelte-ui5-table-problem
Although this problem only exists (as far as I know) in conjunction with Svelte, it would be really kind of you folks if you could check it out. (I'll buy you a beer at next UI5con if you do 😄 ).

**Comment by ilhan007** [comment]
Let "s keep the issue and no problem, we will look into it!

**Comment by ilhan007** [comment]
Hello @brodo

I took a look of the https://github.com/brodo/svelte-ui5-table-problem

When I run npm install and npm run dev and open the app locally,
the app throws with
```js
App.svelte:9 Uncaught ReferenceError: setImmediate is not defined

setImmediate(function () {
  users = [
    { name: "John", username: "john1995", email: "john@example.com"},
    {name: "Carl",username: "cookie123",email: "carl.smith@example.com"}
  ];
});

```

If you declare the users data like this, the ui5-table is properly displayed.
```js
const users = [
  { name: "John", username: "john1995", email: "john@example.com" },
  { name: "Carl",username: "cookie123",email: "carl.smith@example.com" }
];
```

**Comment by pskelin** [comment]
Hi @brodo,

thanks for the beer offer, we might return it for all the good issues you are reporting :)

I found the bug (not sure it was introduced recently, must be the way Svelte creates the DOM). Anyway, our base class does not detect properly that the cells are added to the rows and the rows are effectively empty.

Until we have a properly tested solution (I made a fix that works but don't want to rush it), here is a workaround you can apply in your application to get the table working until the fix is ready.

```diff
setTimeout(function () {
	users = [{ name: "John", username: "john1995", email: "john@example.com"},
		{name: "Carl",username: "cookie123",email: "carl.smith@example.com"}];
}, 0);
+setTimeout(function () {
+	document.querySelectorAll("ui5-table-row").forEach(r => r._updateSlots());
+}, 0);
```

Basically just call the private _updateSlots method on each ui5-table-row after it is created by svelte (you might need to adjust the timeout, 0 worked for me but you can increase it just in case). I also changed the setImmediate to setTimeout, are you using a polyfill for that?

Regards,
Petar

**Comment by brodo** [comment]
Thank you so much! This is really helpful. I don't use a polyfill for setImmediate, it just worked. Maybe there is some transplantation magic going on or Firefox Dev Edition implements it... In my real app, the data comes from a `fetch`.

**Comment by pskelin** [comment]
Thanks for taking the time to report this with the svelte setup @brodo .

You were right that rc.2 was working, then we did some async changes that lead to this bug. Your standalone example was also on the right track. In the end I had to look at the code generated by the svelte compiler, and the order was somewhat different than what someone would write by hand.

```js
insert_dev(target, ui5_table_row, anchor);
append_dev(ui5_table_row, ui5_table_cell0);
append_dev(ui5_table_cell0, t0);
append_dev(ui5_table_row, t1);
append_dev(ui5_table_row, ui5_table_cell1);
append_dev(ui5_table_cell1, t2);
append_dev(ui5_table_row, t3);
append_dev(ui5_table_row, ui5_table_cell2);
append_dev(ui5_table_cell2, t4);
append_dev(ui5_table_row, t5);
```

The problem was when the second line adds the row to the table, and should now be fixed. Please use the suggested workaround until the next release.
